File: completed-labs/14/cpp/GraphVertexTest.cpp
#include <cppunit/extensions/HelperMacros.h>

#include "Graph.hpp"

class GraphVertexTest : public CppUnit::TestFixture {
  CPPUNIT_TEST_SUITE(GraphVertexTest);
  CPPUNIT_TEST(testNoVertices);
  CPPUNIT_TEST(testNoFoundVertices);
  CPPUNIT_TEST(testVertices);
  CPPUNIT_TEST_SUITE_END();

public:
  void setUp() { };
  void tearDown() { };

  void testNoVertices();
  void testNoFoundVertices();
  void testVertices();
};

void GraphVertexTest::testNoVertices() {
  Graph g;
  CPPUNIT_ASSERT_EQUAL_MESSAGE("fresh graph has vertices (or NULL vlist)", 0UL, (unsigned long) g.vlist.size());
}

void GraphVertexTest::testNoFoundVertices() {
  Graph g;
  CPPUNIT_ASSERT_MESSAGE("fresh graph has non-null getVertex(A)", !g.getVertex("A"));
}

void GraphVertexTest::testVertices() {
  Graph g;
  g.addVertex("A");
  g.addVertex("B");
  Vertex *v = g.getVertex("A");
  CPPUNIT_ASSERT_MESSAGE("getVertex(A) fails to find vertex", v);
  CPPUNIT_ASSERT_EQUAL_MESSAGE("getVertex(A) finds vertex with the wrong name", std::string("A"), v->name);
  v = g.getVertex("B");
  CPPUNIT_ASSERT_MESSAGE("getVertex(B) fails to find vertex", v);
  CPPUNIT_ASSERT_EQUAL_MESSAGE("getVertex(B) finds vertex with the wrong name", std::string("B"), v->name);
}

CPPUNIT_TEST_SUITE_NAMED_REGISTRATION(GraphVertexTest, "Vertex");
CPPUNIT_REGISTRY_ADD_TO_DEFAULT("Vertex");

File: is52038b-labs/14/java/GraphVertexTest.java
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;
import java.util.Random;
import org.junit.Test;
import org.junit.runner.JUnitCore;
import org.junit.runner.Result;

public class GraphVertexTest {
    @Test
    public void TestNoVertices() {
	Graph g = new Graph();
	assertEquals("fresh graph has vertices (or null vlist)", 0, g.vlist.size());
    }
    @Test
    public void TestNoFoundVertices() {
	Graph g = new Graph();
	assertEquals("fresh graph has non-null getVertex(A)", null, g.getVertex("A"));
    }
    @Test
    public void TestVertices() {
	Graph g = new Graph();
	g.addVertex("A");
	g.addVertex("B");
	Vertex v = g.getVertex("A");
	assertNotEquals("getVertex(A) fails to find vertex", null, v);
	assertEquals("getVertex(A) finds vertex with the wrong name", "A", v.name);
	v = g.getVertex("B");
	assertNotEquals("getVertex(B) fails to find vertex", null, v);
	assertEquals("getVertex(B) finds vertex with the wrong name", "B", v.name);
    }
}
